### PR TITLE
Update Developer Guide for v1.3 

### PR DESCRIPTION
**Pull request by yashpola**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -158,103 +158,67 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+* `Applicant#updateCurrentStatus()` — Updates the `currentStatus` of the applicant to "pending interview".
+* `Applicant#revertCurrentStatus()` — Reverts the `currentStatus` of the applicant to `previousStatus`.
+* `Interviewer#updateCurrentStatus()` — Updates the `currentStatus` of the interviewer to "interview with [applicant name]" if the `currentStatus` is "free", and vice versa.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+The following class diagram shows the structure of `AddApplicantStatusCommand`, `AddInterviewerStatusCommand`, `ApplicantStatus`, `InterviewerStatus`:
 
-<puml src="diagrams/UndoRedoState0.puml" alt="UndoRedoState0" />
+<puml src="diagrams/add-status/StatusCommandClasses.puml" width="250"/>
```
Comment (macareonie) [inline docs/DeveloperGuide.md:180]: Good effort with including a UML diagram to illustrate your implementation

**Review by macareonie — APPROVED** [review]
👍

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -158,103 +158,67 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
-### \[Proposed\] Undo/redo feature
+### Applicant/Interviewer status feature
```
Comment (headcube1) [inline docs/DeveloperGuide.md:161]: Great feature added, very helpful towards hiring managers!

**Review by headcube1 — APPROVED** [review]
LGTM

**Pull request merged**